src/scheduler/stat.rs

```diff
@@ -3,6 +3,13 @@ use super::work_counter::{WorkCounter, WorkCounterBase, WorkDuration};
 use std::any::TypeId;
 use std::collections::HashMap;
 use std::sync::atomic::{AtomicBool, Ordering};
+#[cfg(feature = "perf")]
+use std::sync::RwLock;
+
+#[cfg(feature = "perf")]
+lazy_static! {
+    pub static ref PERF_EVENTS: RwLock<Vec<(String, i32, i32)>> = RwLock::new(vec![]);
```

Comment:
It seems unnecessary to have this global to temporarily store the parsed events. You could simply have a `PerfEventOptions(Vec<PerfEventConfig>)` in `Options`, and implement `FromStr` for the type. When we process options, the perf event option will be parsed and saved. When creating `WorkerLocalStats`, you just need a reference to `Options` or `MMTK` to access the options.

Reply:
`validate_perf_events` and `parse_perf_events` can be all put to the new `PerfEventOptions` type.